cpp/include/rapidsmpf/buffer/packed_data.hpp

```diff
@@ -22,39 +24,39 @@ namespace rapidsmpf {
  */
 struct PackedData {
     std::unique_ptr<std::vector<std::uint8_t>> metadata;  ///< The metadata
-    std::unique_ptr<rmm::device_buffer> gpu_data;  ///< The gpu data
+    std::unique_ptr<Buffer> gpu_data;  ///< The gpu data
+    /// @brief Default constructor
+    PackedData() = default;
 
     /**
-     * @brief Construct packed data from metadata and gpu data, taking ownership.
+     * @brief Construct from metadata and gpu data.
```

Comment:
```suggestion
     * @brief Construct from metadata and gpu data, taking ownership.
```
This still takes ownership of the provided objects.